[배의진] Week3

BaeUiJin opened the pull request:
## ○ 기본 요구사항

<랜딩 페이지 - og 메타 태그 >

- [x]  페이스북, 카카오톡, 디스코드, 트위터 등 SNS에서 Linkbrary 랜딩 페이지(“/”) 공유 시 미리보기를 볼 수 있도록 랜딩 페이지 메타 태그를 설정했나요?
- [x]  미리보기에서 제목은 “Linkbrary”, 설명은 “세상의 모든 정보를 쉽게 저장하고 관리해 보세요”로 설정했나요?
- [x]  주소와 이미지는 자유롭게 설정했나요?

<랜딩 페이지>

- [x]  브라우저에 현재 보이는 화면의 영역(viewport) 너비를 기준으로 분기되는 반응형 디자인을 적용했나요?
    
    PC: 1200px 이상
    Tablet: 768px 이상 ~ 1199px 이하
    Mobile: 375px 이상 ~ 767px 이하
    375px 미만 사이즈의 디자인은 고려하지 않습니다
    
- [x]  Tablet 사이즈에서 화면의 너비가 1199px 이하로 작아질 때 “Linkbrary” 로고와 “로그인” 버튼 사이의 간격은 변하지 않게 고정값으로 유지하되 좌우 여백이 줄어드나요?
- [x]  Tablet 사이즈에서 최소 좌우 여백은 “Linkbrary” 로고의 왼쪽에 여백 32px, “로그인” 버튼 오른쪽 여백 32px을 유지하고“Linkbrary” 로고와 “로그인" 버튼의 간격이 가까워지나요?
- [x]  PC, Tablet 사이즈의 이미지 크기는 고정값을 사용했나요?
- [x]  Mobile 사이즈의 이미지는 좌우 여백 32px을 제외하고 이미지 영역이 꽉 차나요?
(이때 가로가 커지는 비율에 맞춰 세로도 커져야 합니다.)
- [x]  Mobile 사이즈 너비가 커지면, “Privacy Policy”, “FAQ”, “codeit-2023”이 있는 영역과 SNS 아이콘들이 있는 영역의 간격이 커지나요?

<로그인, 회원가입 페이지>

- [x]  Tablet 사이즈에서 디자인은 PC사이즈와 동일한가요?
- [x]  Mobile 사이즈에서 좌우 여백 32px 제외하고 내부 요소들이 너비를 모두 차지하나요?
- [x]  Mobile 사이즈에서 내부 요소들의 너비는 기기의 너비가 커지는 만큼 커지지만 400px을 넘지 않나요?

---
## ○ 심화 요구사항

- [ ]  랜딩 페이지 Mobile 사이즈에서 제품 소개 영역의 순서를 제목, 설명, 이미지 => 제목, 이미지, 설명 순서로 변경했나요?

---
## ○ 질문

❓PC 에서 Chrome 으로 모바일 버전의 코딩 결과를 출력하려고 하는데, 창 너비가 500px 밑으로 줄여지지 않습니다. 코딩이 잘못되어서 그런걸까요?



❓<landing.html> 안에 <heading-imge> 클래스가 적용된 이미지는 모바일 버전에서는 창 크기가 줄어들면서 이미지 크기가 줄어듦과 동시에 가로 세로 비율을 유지하여야 한다고 합니다. 겨우 구현하긴 했지만, 뭔가 지저분하고 좋지 않은 코드인 것 같습니다. 더 효율적인 방법이 있을까요~?

![Untitled (7)](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/160081323/66989133-18dd-4e37-9943-bdb64daf2e04)




❓위 심화 문제는 (1)을 (3)으로 바꾸는 건데, 어떻게 코딩을 해야 할지 모르겠습니다. 아래 첨부한 코드는 그림 (2)의 코드입니다!!

(1) tablet version                                              
![Untitled (4)](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/160081323/e9b1cd98-90b1-4629-b66f-4b5f0d4c914a)

(2) mobile version         
![Untitled (5)](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/160081323/5f805ba4-b0c7-4ca7-a948-d8c544f76f06)


(3) mobile version 
![Untitled (6)](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/160081323/ce992820-2c4e-481b-87e9-3ab976499a73)

#



html

```html
<section class="contents">
  <div>
    <h2 class="contents-text-title">
      <span class="contents-text-gradient first">원하는 링크</span>를<br class="break-not-mobile">
      저장하세요
    </h2>
    <p class="contents-text-small">
      나중에 읽고 싶은 글, 다시 보고 싶은 영상,<br class="break-not-mobile">
      사고 싶은 옷, 기억하고 싶은 모든 것을<br class="break-not-mobile">
      한 공간에 저장하세요.
    </p>
  </div>
  <div class="contents-image" alt="이미지-링크" style="background-image: url(./images/img1.svg);">
  </div>
</section>
```

css

```css
.contents {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

/*tablet*/
@media (min-width: 768px) and (max-width: 1199px) {
	.contents {
	  max-width: 700px;
	  gap: 51px;
	  align-items: center;
	  padding: 80px 0 50px;
	}
}

/*mobile*/
@media (max-width: 767px) {
	.contents {
	  flex-direction: column;
	  gap: 24px;
	  align-items: left;
	  padding: 40px 32px 36px;
	}
}
```




❓ 그림은 <landing.html> 의<footer> 태그에 해당하는 부분입니다. 가이드를 보면 여기에는 64px 만큼 높이가 있는데, 이 부분에 height 값을 주어도 어째서인지 적용되질 않는데 이유를 잘 모르겠습니다.

(가이드)
![Untitled (7)](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/160081323/e3952d82-ca33-4126-b5d1-bfbe3bf293fd)

html

```html
<footer>
<section class="info">
  <div class="info-box">
    <div class="info-author">
      ©codeit - 2023
    </div>
    <div>
      <div class="info-description">
        <a href="./privacy/Privacy.html">Privacy Police</a>
        <a href="./faq/Faq.html">FAQ</a>
      </div>
    </div>
  </div>
  <div class="info-sns">
    <a href="">
      <img alt="facebook" src="./images/icon_facebook.svg">
    </a>
    <a href="">
      <img alt="twitter" src="./images/icon_twitter.svg">
    </a>
    <a href="">
      <img alt="youtube" src="./images/icon_youtube.svg">
    </a>
    <a href="">
      <img alt="instagram" src="./images/icon_instagram.svg">
    </a>
  </div>
</section>
```

css

```css
.info {
display: flex;
/* height 값이 적용되지 않습니다. */
height: 120px;
align-items: flex-start;
justify-content: space-between;
padding: 32px 104px 64px;
}
```

BaeUiJin committed:
Renamed 5 files

executing git status command, korean file names were translated into unicode.

thus, unable to identify file with its name.

solution : renamed into english. problem solved

BaeUiJin committed:
Changed file names of hyperlinks

changed file names of hyperlinks of href property as files were renamed into english

BaeUiJin committed:
Deleted png files

BaeUiJin committed:
Replaced 'button' tag selector with class selctor

tag selectors are less specific, harder to debug than class selectors.

thus, reassigned buttons of signin and signup page with class selectors.

BaeUiJin committed:
Deleted meta tags

BaeUiJin committed:
Replace img-main.png with right one

wrong image file has been placed as img-main.png, thus replaced with
right one.

plus, changed image type from png to svg to solve clearity issue when
image size get small.

"head" class's width is adjusted to the same one with "content" class's
width for consistency. 100% to 1050px.

BaeUiJin committed:
Main image in landing.html adjusted

image size, position is changed as instruction.
'box-sizing'option set as 'content-box' and applied globally.

BaeUiJin committed:
Four images in <main>tag in landing.html adjusted

images imported at <div> tag via CSS style rather than <img> tag to fix
its size.

BaeUiJin committed:
Add margin to nav logo and button

BaeUiJin committed:
Add margin to nav logo and button

BaeUiJin committed:
Changed landing page's main image padding to margin

BaeUiJin committed:
Tablet mode set

BaeUiJin committed:
Add Mobile landing page

BaeUiJin committed:
Add signin & signup page for mobile

BaeUiJin committed:
Merge branch 'tablet' into release

BaeUiJin committed:
Merge branch 'mobile' into release

BaeUiJin committed:
Add og tag draft

Siihyun commented on signin/style-signin.css:
```diff
@@ -106,4 +111,11 @@ button {
+
+@media (min-width: 767px) {
+  section {
+    margin: 120px auto 233px;
```
margin으로 말고, flex, justtify-content, align-item을 통해서 중앙정렬 해보는건 어떨까요?

Siihyun commented on signup/style-signup.css:
```diff
@@ -4,15 +4,20 @@ body {
-  width: 400px;
-  margin-top: 238px;
-  /* border: 2px solid red; */
+  box-sizing: content-box;
```
box-sizing: content-box의 경우 global style에 전체적으로 적용해주는게 좋습니다.
전체적으로 적용되어야 하는 속성이기 때문입니다.

Siihyun commented on signup/style-signup.css:
```diff
@@ -4,15 +4,20 @@ body {
-  margin-top: 238px;
-  /* border: 2px solid red; */
+  box-sizing: content-box;
+  margin: 200px auto 250px;
```
레이아웃을 대부분 마진으로 해결하시려는 경향이 있는 것 같습니다.
margin의 경우 특정한 수치를 적용해야 하기 때문에 유연하지 않습니다.

전체적인 레이아웃을 잡는 경우(예 : 화면 중앙에 배치) display: flex와 같은 값을 쓰는게 좋습니다.

Siihyun reviewed (approved):
고생하셨습니다! 
다음 과제부터는 다른 분들처럼 netlify 링크도 같이 전달 부탁드려요!

질문에 대한 답변은 별도 커멘트로 드릴게요.

Siihyun commented:
❓PC 에서 Chrome 으로 모바일 버전의 코딩 결과를 출력하려고 하는데, 창 너비가 500px 밑으로 줄여지지 않습니다. 코딩이 잘못되어서 그런걸까요?

=> 요거 이해를 못했어요. 전 잘 되는걸요? ㅎㅎ 커멘트로 추가 설명해주시면 답변드려볼게요.

![image](https://github.com/codeit-bootcamp-frontend/5-Weekly-Mission/assets/44644821/14b41991-7d04-4e55-88c8-64784cb613ce)


---

❓<landing.html> 안에 클래스가 적용된 이미지는 모바일 버전에서는 창 크기가 줄어들면서 이미지 크기가 줄어듦과 동시에 가로 세로 비율을 유지하여야 한다고 합니다. 겨우 구현하긴 했지만, 뭔가 지저분하고 좋지 않은 코드인 것 같습니다. 더 효율적인 방법이 있을까요~?

=> css의 aspect-ratio 속성을 한번 살펴보세요!

---

❓위 심화 문제는 (1)을 (3)으로 바꾸는 건데, 어떻게 코딩을 해야 할지 모르겠습니다. 아래 첨부한 코드는 그림 (2)의 코드입니다!!

=> flex의 order 속성을 한번 찾아보세요!

---

❓ 그림은 <landing.html> 의

태그에 해당하는 부분입니다. 가이드를 보면 여기에는 64px 만큼 높이가 있는데, 이 부분에 height 값을 주어도 어째서인지 적용되질 않는데 이유를 잘 모르겠습니다.

=> footer쪽에 margin-top: 100px이 적용되어 있기 때문입니다! 지우고 시도해보시죠!

---

BaeUiJin committed:
Changed signin page

To align items center of the browser, applied 'flex' instead of 'margin'

BaeUiJin committed:
Changed signup page

To align element at the center of the page, applied 'flex' instead of
'margin'

BaeUiJin committed:
Changed landing page

1.in 'head-image' class tag, replaced 'padding-top' with 'aspect-ratio' to make
  code right.

2.in 'contents' class tag, changed order of elements with 'order'
  property.

3.fixed height issue of the 'info' class tag in the 'footer'tag.

BaeUiJin commented:
멘토님께서 달아주신 코멘트 모두 반영했습니다!

1. 객체를 화면에 가운데 정렬할 때 `margin` 대신 `flex, justtify-content=center, align-item=center, min-height=dvh`를 적용했습니다.
2. `box-sizing` 프로퍼티는 * {} 으로 문서 전체에 일괄 적용했습니다.
3. landing page 에서 `width=100%, aspect-ratio= 1 / 0.55` 를 적용해 모바일 버전일 때 head-image 비례를 유지하며 늘어나고 줄어들도록 했습니다. 그 전에는 `padding-top=55%` 를 적용했었습니다.  
4. landing page 에서 `order`와 `display=contents` 프로퍼티를 활용해 요소 간 순서를 조정하였습니다.
5. landing page 에서 footer 안 'info' 클래스 div 태그에 `height=64px` 가 적용되지 않는 문제 해결되었습니다.
